src/libs/actions/OnyxUpdateManager/index.ts

```diff
@@ -38,17 +39,40 @@ Onyx.connect({
     },
 });
 
+let queryPromise: Promise<Response | Response[] | void> | undefined;
+
+let resolveQueryPromiseWrapper: () => void;
+const createQueryPromiseWrapper = () =>
+    new Promise<void>((resolve) => {
+        resolveQueryPromiseWrapper = resolve;
+    });
+// eslint-disable-next-line import/no-mutable-exports
+let queryPromiseWrapper = createQueryPromiseWrapper();
+
+const resetDeferralLogicVariables = () => {
+    queryPromise = undefined;
+    deferredUpdatesProxy.deferredUpdates = {};
```

Comment:
I think we don't need to reset here. We only need to reset it in method `validateAndApplyDeferredUpdates`